Sources/ProjectDescription/Headers.swift

```diff
@@ -64,4 +70,58 @@ public struct Headers: Codable, Equatable {
             exclusionRule: exclusionRule
         )
     }
+
+    private static func headers(from list: FileList,
+                                umbrella: Path,
+                                private: FileList? = nil,
+                                allOthersAsProject: Bool) -> Headers
+    {
+        .init(
+            public: list,
+            umbrellaHeader: umbrella,
+            private: `private`,
+            project: allOthersAsProject ? list : nil,
+            exclusionRule: .projectExcludesPrivateAndPublic
+        )
+    }
+
+    /// Loading headers from the file list,
+    /// as `public` will be marked all presented in the umbrella header
+    /// as `private` - all from the file list of private headers (exclude already found `public`)
+    /// as `project` - all others
+    /// - Parameters:
+    ///     - from: File list, which contains `public` and `project` headers
+    ///     - umbrella: File path to the umbrella header
+    ///     - private: File list, which contains `private` headers
+    public static func allHeaders(from list: FileList,
+                                  umbrella: Path,
+                                  private: FileList? = nil) -> Headers
+    {
+        headers(
+            from: list,
+            umbrella: umbrella,
+            private: `private`,
+            allOthersAsProject: true
+        )
+    }
```

Comment:
nit: just to avoid having to use "`", same below
```suggestion
                                  private privateHeaders: FileList? = nil) -> Headers
    {
        headers(
            from: list,
            umbrella: umbrella,
            private: privateHeaders,
            allOthersAsProject: true
        )
    }
```